src/store/GlobalState.tsx

```diff
@@ -0,0 +1,44 @@
+import { createContext, useEffect, useState } from 'react';
+import { collection, query, onSnapshot } from 'firebase/firestore';
+
+import db from '../firebase';
+
+import { Expense } from '../expense';
+
+interface GlobalState {
+  expenses: Expense[];
+}
+
+const initialState: GlobalState = { expenses: [] };
+
+export const GlobalContext = createContext(initialState);
```

Comment:
the `GlobalContext` is quite general and it does not really speak to its contents. can we think of a better name?